lib/portlayer/storage/image_cache.go

```diff
@@ -232,6 +232,18 @@ func (c *NameLookupCache) WriteImage(op trace.Operation, parent *Image, ID strin
 	return i, nil
 }
 
+func (c *NameLookupCache) Read(op trace.Operation, store *url.URL, child, ancestor string, pathspec map[string]string, data bool) (io.ReadCloser, error) {
+	_, err := util.ImageStoreName(store)
+	if err != nil {
+		return nil, err
+	}
+
+	if archiver, ok := c.DataStore.(Archiver); ok {
+		return archiver.Read(op, store, child, ancestor, pathspec, data)
```

Comment:
I'm confused. The code type assert the c to Archiver which means c implements the interface. We now that c's type is datastore so why we are passing something to Read? Read is already implemented on Store type and I'm guessing Store knows/keeps its own URL somewhere?